cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -870,11 +870,18 @@ func (a *StaticAutoscaler) deleteCreatedNodesWithErrors() bool {
 			klog.Warningf("Error while trying to delete nodes from %v: %v", nodeGroupId, err)
 		} else {
 			deletedAny = true
-			a.clusterStateRegistry.InvalidateNodeInstancesCacheEntry(nodeGroup)
+			err = a.clusterStateRegistry.HandleDeletingFailedScaleUpNodes(nodeGroup)
+			if err != nil {
+				returnErr = fmt.Errorf("Failed to clean up state after deleting failed nodes from %s node group: %v", nodeGroup.Id(), err)
+				klog.Warningf(returnErr.Error())
```

Comment:
I'd log this on Error() level - I'd argue any error that causes us to break the loop should be flagged as a serious problem

Reply:
Actually, Recalculate() doesn't return that error, so we're no longer breaking the loop after recent changes. Do you think that will cause issues?